src/GeometryEvaluator.cc

```diff
@@ -74,27 +162,44 @@ shared_ptr<const Geometry> GeometryEvaluator::evaluateGeometry(const AbstractNod
 		smartCacheInsert(node, this->root);
```

Comment:
This is modifying the cache outside of lock

Reply:
Good catch, thanks. I'll get this fixed.

Reply:
Fixed in 59712015f379403abe3fc711e3e1854ed7e8ddab